lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1662,6 +1653,38 @@ void NVMMASharedEncodingAttr::print(AsmPrinter &printer) const {
   printer << "}>";
 }
 
+int NVMMASharedEncodingAttr::getVec() const {
+  if (getSwizzlingByteWidth() == 0)
+    return 1;
+  return 128 / getElementBitWidth();
+}
+
+int NVMMASharedEncodingAttr::getPerPhase() const {
+  if (getSwizzlingByteWidth() == 0)
+    return 1;
+  return 128 / getSwizzlingByteWidth();
+}
```

Comment:
Shouldn't this always `return 1`? At least in theory. I think the change that @ThomasRaoux added in https://github.com/triton-lang/triton/pull/6482 was basically an "automatic perPhase inference depending on shape" of sorts.

Reply:
I meant the one regarding packing different dimensions into the swizzling dimension.
But yeah, I had misread, I guess you need this to tile the 8x8 core tile. I got confused by the PTX docs.